controllers/apim/ratelimitpolicy_controller.go

```diff
@@ -136,100 +145,113 @@ func (r *RateLimitPolicyReconciler) Reconcile(eventCtx context.Context, req ctrl
 
 func (r *RateLimitPolicyReconciler) reconcileWithVirtualService(ctx context.Context, vs *istio.VirtualService, rlp *apimv1alpha1.RateLimitPolicy) error {
 	logger := r.Logger()
+	rlpKey := client.ObjectKeyFromObject(rlp)
 
 	_, ok := vs.Annotations[VSAnnotationRateLimitPolicy]
 	if !ok {
-		rlpNamespacedName := client.ObjectKeyFromObject(rlp)
-		vs.Annotations[VSAnnotationRateLimitPolicy] = rlpNamespacedName.String()
+		vs.Annotations[VSAnnotationRateLimitPolicy] = rlpKey.String()
 		if err := r.Client().Update(ctx, vs); err != nil {
-			logger.Error(err, "failed to add ratelimitpolicy annotation to virtualservice")
+			logger.Error(err, "failed to add RateLimitPolicy annotation to VirtualService")
 			return err
 		}
-		logger.V(1).Info("successfully added ratelimitpolicy annotation to virtualservice")
+		logger.V(1).Info("successfully added RateLimitPolicy annotation to VirtualService")
 	}
 
 	// TODO(rahulanand16nov): store context of virtualservice in RLP's status block and manage envoy patches.
-	if err := r.reconcileRateLimitFilters(ctx, vs, rlp); err != nil {
-		logger.Error(err, "failed to reconcile ratelimit filters")
-		return err
-	}
 
-	if err := r.reconcileRateLimitActions(ctx, vs, rlp); err != nil {
-		return err
-	}
-	logger.Info("successfully reconciled ratelimitpolicy using attached virtualservice")
-	return nil
-}
+	// create/update EnvoyFilter patches for each gateway
+	for _, gw := range vs.Spec.Gateways {
+		gwKey := common.NamespacedNameToObjectKey(gw, vs.Namespace) // Istio defaults to VirtualService's namespace
```

Comment:
not sure I understand this. What is meant by Istio defaults to VirtualService's namespace?

Reply:
Gateways are referenced in the `<namespace>/<name>` format but if the namespace is not mentioned, Istio will look into VirtualService's namespace.